### PR TITLE
[WIP] Update to scala 2.13, add scalafix, add scalafmt, update travis.yml.

### PR DESCRIPTION
**Pull request by dataswifty**
WIP as it depends on the hat-client.
Update to scala 2.13, add scalafix, add scalafmt, update travis.yml.

### CODE REVIEW

**Comment by dataswifty** [comment]
The CI is failing because hat-client v2.6.13 for Scala 2.13 has not been published yet.

File: build.sbt [hunk 1]
```diff
@@ -25,3 +25,12 @@ publishTo := {
...
 }
+
+inThisBuild(
+  List(
+    scalaVersion := "2.13.3",
+    scalafixScalaBinaryVersion := "2.13",
+    semanticdbEnabled := true,
+    semanticdbVersion := scalafixSemanticdb.revision
+  )
+)
```
Comment (whiteshadow-gr) [inline build.sbt:36]: New line please

File: project/BuildSettings.scala [hunk 2]
```diff
@@ -71,18 +71,5 @@ object BasicSettings extends AutoPlugin {
...
-
-  lazy val scalariformPrefs = Seq(
-    ScalariformKeys.preferences := ScalariformKeys.preferences.value
-      .setPreference(FormatXml, false)
-      .setPreference(DoubleIndentClassDeclaration, true)
-      .setPreference(DoubleIndentConstructorArguments, true)
-      .setPreference(AlignSingleLineCaseStatements, true)
-      .setPreference(CompactControlReadability, true)
-      .setPreference(DanglingCloseParenthesis, Prevent))
+    )
```
Comment (whiteshadow-gr) [inline project/BuildSettings.scala:74]: why did we remove the scalariform? In general I am a lost with the linters and the tools for backend

File: src/main/scala/org/hatdex/dex/api/json/DexJsonFormats.scala [hunk 3]
```diff
@@ -17,14 +17,14 @@ trait DexJsonFormats extends HatJsonFormats {
   import play.api.libs.json.JodaWrites._
   import play.api.libs.json.JodaReads._
 
-  implicit val offerHatFormat            = Json.format[OfferHat]
-  implicit val offerclaimFormat          = Json.format[OfferClaim]
-  implicit val offerHatCredentialsFormat = Json.format[OfferHatCredentials]
-  implicit val offerClaimsInfoFormat     = Json.format[OfferClaimsInfo]
+  implicit val offerHatFormat: OFormat[OfferHat]            = Json.format[OfferHat]
```
Comment (whiteshadow-gr) [inline src/main/scala/org/hatdex/dex/api/json/DexJsonFormats.scala:20]: I see the formatting is wrong here. I noticed that the `sbt scalafmt` is not running as part of the sbt build command. I need to manually call it before each PR. Ben said he will take a look. Just FYI

File: src/main/scala/org/hatdex/dex/api/models/Notice.scala [hunk 4]
```diff
@@ -11,6 +11,7 @@ package org.hatdex.dex.api.models
 
 import org.joda.time.DateTime
 import play.api.libs.json.Json
+import play.api.libs.json.{ OWrites, Reads }
```
Comment (whiteshadow-gr) [inline src/main/scala/org/hatdex/dex/api/models/Notice.scala:14]: What is the `OWrites` ?? 🤯

**Review by whiteshadow-gr — APPROVED** [review]

File: project/BuildSettings.scala (hunk 2 above)
Comment (dataswifty) [reply project/BuildSettings.scala:74]: Scalariform and scalafmt do the same thing.  And we have settled on scalafmt.

File: src/main/scala/org/hatdex/dex/api/models/Notice.scala (hunk 4 above)
Comment (dataswifty) [reply src/main/scala/org/hatdex/dex/api/models/Notice.scala:14]: https://www.playframework.com/documentation/2.6.13/api/scala/index.html#play.api.libs.json.OWrites

Converts Objects to JsValues.

File: src/main/scala/org/hatdex/dex/api/json/DexJsonFormats.scala (hunk 3 above)
Comment (dataswifty) [reply src/main/scala/org/hatdex/dex/api/json/DexJsonFormats.scala:20]: It's the ordering of scalafix and scalafmt.

File: project/BuildSettings.scala [hunk 5]
```diff
@@ -54,10 +54,10 @@ object BasicSettings extends AutoPlugin {
         "-unchecked", // Enable additional warnings where generated code depends on assumptions.
         "-Xfatal-warnings", // Fail the compilation if there are any warnings.
         "-Xlint", // Enable recommended additional warnings.
-        "-Ywarn-adapted-args", // Warn if an argument list is modified to match the receiver.
+        //"-Ywarn-adapted-args", // Warn if an argument list is modified to match the receiver.
```
Comment (benshaw) [inline project/BuildSettings.scala:57]: Howcome these are removed ?

Comment (dataswifty) [reply project/BuildSettings.scala:57]: They are not supported in 2.13, I did not remove them by accident.

**Review by benshaw — APPROVED** [review]

**Pull request merged**